articles/basics-of-the-object-model/index.md

```diff
@@ -0,0 +1,106 @@
+
+Class (Car Brand) = Objects (Toyota, Subaru, Hyundai, Audi, Volkswagen)
+
+An Object model uses various diagrams to show how objects behave and perform real-world tasks. The diagrams used include use-case diagram, sequence diagram, class diagram, activity diagram, etc.
+
+The object model describes objects in Object-oriented programming, Object-oriented Analysis, and Object-oriented design.
+
+### Understand OOP, OOD, and OOA
+
+#### Object-Oriented Programming (OOP)
+OOP is a programming paradigm with the concept of classes and objects. OOP incorporates properties/attributes and functions/methods into an element called an object. These objects are grouped into classes.
+
+This programming style generally exists in programming languages such as C++, Java, JavaScript, Python, etc. These languages help structure and organize systems and software programs quickly. Developers use Object-Oriented Languages when creating large, complex, and actively updated programs.
+
+OOP is based on four principles, Encapsulation, Abstraction, Inheritance, and Polymorphism.
+
+- [Encapsulation](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)): It is the bundling of data and methods into a single unit or a class. It is also known as information/data hiding. In other words, the data is not available to the outside world. Therefore, only those methods that are within the class can access it.
+- [Abstraction](https://stackify.com/oop-concept-abstraction/#:~:text=Abstraction%20is%20one%20of%20the,unnecessary%20details%20from%20the%20user.&amp;text=That&#39;s%20a%20very%20generic%20concept,everywhere%20in%20the%20real%20world.): It hides implementation details while presenting the attributes to the outer world. It reduces code complexity, hides the facts, and expose the essential parts.
+
+For example;
+
+A database system details how a database is created while hiding how data is stored and maintained. The user only sees the database results on the screen.
+
+- [Inheritance](https://en.wikipedia.org/wiki/Inheritance_(object-oriented_programming)): It is a technique of creating a new class from an existing one. A sub-class derives features from the parent class, also known as a base class.
+
+There are various [types of inheritance in OOP](https://en.wikipedia.org/wiki/Inheritance_(object-oriented_programming)):
+
+- Single Inheritance
+- Multiple Inheritance
+- Hierarchical Inheritance
+- Multilevel Inheritance
+- Hybrid Inheritance.
+
+- [Polymorphism](https://www.tutorialspoint.com/java/java_polymorphism.htm#:~:text=Polymorphism%20is%20the%20ability%20of,to%20a%20child%20class%20object.&amp;text=In%20Java%2C%20all%20Java%20objects,and%20for%20the%20class%20Object.): The term polymorphism refers to the occurrence of something in many forms. Objects can be represented in many forms.
+
+For example,
+
+The world has many continents. Each continent consists of many countries, and each state is made of cities.
```

Comment:
I don't really see how the example explains what polymorphism is. Consider using a different one.